File: backend/app/models/emergency_contact.py
import uuid
from datetime import datetime, timezone
from sqlalchemy import Column, String, Boolean, DateTime, ForeignKey
from sqlalchemy.orm import relationship
from app.core.database import Base


class EmergencyContact(Base):
    __tablename__ = "emergency_contacts"

    id = Column(String(36), primary_key=True, default=lambda: str(uuid.uuid4()))
    user_id = Column(String(36), ForeignKey("users.id", ondelete="CASCADE"), nullable=False, index=True)
    name = Column(String(100), nullable=False)
    phone_number = Column(String(30), nullable=False)
    relationship_type = Column(String(50), nullable=False)  # 'spouse', 'parent', 'sibling', 'friend', 'riding_buddy'
    is_verified = Column(Boolean, default=True, nullable=False)
    notify_on_crash = Column(Boolean, default=True, nullable=False)
    notify_on_sos = Column(Boolean, default=True, nullable=False)
    created_at = Column(DateTime(timezone=True), default=lambda: datetime.now(timezone.utc), nullable=False)
    updated_at = Column(DateTime(timezone=True), default=lambda: datetime.now(timezone.utc), onupdate=lambda: datetime.now(timezone.utc), nullable=False)

    user = relationship("User", back_populates="emergency_contacts")

File: mobile/lib/models/emergency_contact.dart
class EmergencyContactModel {
  final String id;
  final String name;
  final String phoneNumber;
  final String relationshipType;
  final bool isVerified;
  final bool notifyOnCrash;
  final bool notifyOnSos;

  EmergencyContactModel({
    required this.id,
    required this.name,
    required this.phoneNumber,
    required this.relationshipType,
    required this.isVerified,
    required this.notifyOnCrash,
    required this.notifyOnSos,
  });

  factory EmergencyContactModel.fromJson(Map<String, dynamic> json) {
    return EmergencyContactModel(
      id: json['id'] ?? '',
      name: json['name'] ?? '',
      phoneNumber: json['phone_number'] ?? '',
      relationshipType: json['relationship_type'] ?? 'friend',
      isVerified: json['is_verified'] ?? true,
      notifyOnCrash: json['notify_on_crash'] ?? true,
      notifyOnSos: json['notify_on_sos'] ?? true,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'phone_number': phoneNumber,
      'relationship_type': relationshipType,
      'notify_on_crash': notifyOnCrash,
      'notify_on_sos': notifyOnSos,
    };
  }
}
